table/table.go

```diff
@@ -72,6 +76,64 @@ func (t Table) LocationProvider() (LocationProvider, error) {
 	return LoadLocationProvider(t.metadataLocation, t.metadata.Properties())
 }
 
+func (t Table) NewTransaction() *Transaction {
+	meta, _ := MetadataBuilderFromBase(t.metadata)
+
+	return &Transaction{
+		tbl:  &t,
+		meta: meta,
+		reqs: []Requirement{},
```

Comment:
Why wouldn't you keep the updates and requirements next to each other in the Transaction?

Later on, when we implement retries, the metadata might refresh if a table has been updated. Then we can detect if we can re-apply the changes. Therefore I think it is better to keep this at the Transaction level.